checker/src/main/java/org/checkerframework/checker/pico/jdk.astub

```diff
@@ -0,0 +1,355 @@
+import org.checkerframework.checker.pico.qual.ReceiverDependentMutable;
+import org.checkerframework.checker.pico.qual.Readonly;
+import org.checkerframework.checker.pico.qual.ObjectIdentityMethod;
+import java.util.Collection;
+
+package java.lang;
+
+@ReceiverDependentMutable
+class Iterable<E> {
+  Iterator<E> iterator(@Readonly Iterable this);
+}
+
+class AssertionError {
+  AssertionError(@Readonly Object var0);
+}
+
+class RuntimeException {
+  RuntimeException(String var0, Throwable var1);
+}
+
+@ReceiverDependentMutable
+class Object {
+  @ReceiverDependentMutable Object();
+  Class<?> getClass(@Readonly Object this);
+  String toString(@Readonly Object this);
+  int hashCode(@Readonly Object this);
+  boolean equals(@Readonly Object this, @Readonly Object var1);
+  @ReceiverDependentMutable Object clone(@ReceiverDependentMutable Object this);
+  @ObjectIdentityMethod
+  final native Class<?> getClass();
+}
+
+class String {
+  int length(@Immutable String this);
+  char charAt(@Immutable String this, int var1);
+  String replace(@Readonly CharSequence target, @Readonly CharSequence replacement);
+  boolean contains(@Readonly CharSequence s);
+  String substring(@Immutable String this, int var1);
+  String substring(@Immutable String this, int var1, int var2);
+  String toString(@Immutable String this);
+  boolean equals(@Immutable Object var1);
+  boolean contentEquals(@Readonly CharSequence var1);
+  static String valueOf(@Readonly Object var0);
+  static String format(String var0, @Readonly Object @Readonly ... var1);
+  static String format(@Readonly Locale l, String format, @Readonly Object @Readonly ... var1);
+}
+
+class StringBuilder {
+  StringBuilder append(@Readonly Object var1);
+}
+
+class StringBuffer {
+  int length(@Readonly StringBuffer this);
+  int capacity(@Readonly StringBuffer this);
+  StringBuffer append(@Readonly Object obj);
+  String substring(@Readonly StringBuffer this, int start);
+  CharSequence subSequence(@Readonly StringBuffer this, int start, int end);
+  String substring(@Readonly StringBuffer this, int start, int end);
+  int indexOf(@Readonly StringBuffer this, String str);
+  int indexOf(@Readonly StringBuffer this, String str, int fromIndex);
+  int lastIndexOf(@Readonly StringBuffer this, String str);
+  int lastIndexOf(@Readonly StringBuffer this, String str, int fromIndex);
+}
+
+@ReceiverDependentMutable
+class Throwable {
+  String getMessage(@ReceiverDependentMutable Throwable this);
+  String getLocalizedMessage(@ReceiverDependentMutable Throwable this);
+  Throwable getCause(@ReceiverDependentMutable Throwable this);
+  void printStackTrace(@ReceiverDependentMutable Throwable this);
+  void printStackTrace(@ReceiverDependentMutable Throwable this, PrintStream var1);
+  void printStackTrace(@ReceiverDependentMutable Throwable this, Throwable.PrintStreamOrWriter var1);
+}
+
+@ReceiverDependentMutable
+interface CharSequence {
+  int length(@Readonly CharSequence this);
```

Comment:
I wonder whether the default for `CharSequence` should be readonly... the interface itself must be RDM, e.g. to allow StringBuilder, which is mutable. But CharSequence itself is readonly?
So adding [DefaultQualifierForUse](checker-qual/src/main/java/org/checkerframework/framework/qual/DefaultQualifierForUse.java) to such classes might make their use easier?

Reply:
Done.